test/e2e/da_posting_integration_test.go

```diff
@@ -0,0 +1,333 @@
+			return false
+		}
+		dataNSB64 := base64.StdEncoding.EncodeToString(dataNamespace.Bytes())
+		for _, tx := range res.Txs {
+			if tx.TxResult.Code != 0 {
+				continue
+			}
+			for _, ev := range tx.TxResult.Events {
+				if ev.Type != "celestia.blob.v1.EventPayForBlobs" {
+					continue
+				}
+				for _, attr := range ev.Attributes {
+					if string(attr.Key) == "namespaces" && strings.Contains(string(attr.Value), dataNSB64) {
+						pfbHeight = tx.Height
+						return true
+					}
+				}
+			}
+		}
+		return false
+	}, "expected a PayForBlobs tx on celestia-app")
+
+	waitFor(ctx, t, time.Minute, 5*time.Second, func() bool {
+		if pfbHeight == 0 {
+			return false
+		}
+		for h := pfbHeight; h <= pfbHeight+10; h++ {
+			ids, err := daClient.DA.GetIDs(ctx, uint64(h), dataNamespace.Bytes())
+			if err != nil {
+				t.Logf("GetIDs data height=%d err=%v", h, err)
+				continue
+			}
+			if ids != nil && len(ids.IDs) > 0 {
+				return true
+			}
+		}
+		return false
+	}, "expected blob in DA for namespace ev-data")
+}
+
+// newHTTPClient is a small helper to avoid importing the docker_e2e client.
+func newHTTPClient(host, port string) (*httpClient, error) {
+	return &httpClient{baseURL: fmt.Sprintf("http://%s:%s", host, port)}, nil
+}
+
+type httpClient struct {
+	baseURL string
+}
+
+func (c *httpClient) Get(ctx context.Context, path string) ([]byte, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, c.baseURL+path, nil)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("HTTP %d: %s", resp.StatusCode, resp.Status)
+	}
+	return io.ReadAll(resp.Body)
+}
+
+func (c *httpClient) Post(ctx context.Context, path, key, value string) ([]byte, error) {
+	body := strings.NewReader(fmt.Sprintf("%s=%s", key, value))
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, c.baseURL+path, body)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", "text/plain")
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	return io.ReadAll(resp.Body)
+}
+
+// getEvNodeImage resolves the EV Node image to use for the test.
+// Falls back to EV_NODE_IMAGE_REPO:EV_NODE_IMAGE_TAG or evstack:local-dev.
+func getEvNodeImage() container.Image {
+	repo := strings.TrimSpace(getEnvDefault("EV_NODE_IMAGE_REPO", "evstack"))
+	tag := strings.TrimSpace(getEnvDefault("EV_NODE_IMAGE_TAG", "local-dev"))
+	return container.NewImage(repo, tag, "10001:10001")
+}
+
+func getEnvDefault(key, def string) string {
+	if v := strings.TrimSpace(os.Getenv(key)); v != "" {
+		return v
+	}
+	return def
+}
+
+// waitFor polls condition until it returns true, context is cancelled, or timeout expires.
+func waitFor(ctx context.Context, t *testing.T, timeout, interval time.Duration, condition func() bool, msg string) {
```

Comment:
this looks like a re-implementation of `wait.ForCondition` which exists in tastora